test/integration/plugins/nuagevsp/nuageTestCase.py

```diff
@@ -461,22 +547,27 @@ def create_NetworkAclRule(self, rule, traffic_type="Ingress", network=None, acl_
                                      )
 
     # ssh_into_VM - Gets into the shell of the given VM using its public IP
-    def ssh_into_VM(self, vm, public_ip, reconnect=True):
-        self.debug("SSH into VM with ID - %s on public IP address - %s" % (vm.id, public_ip.ipaddress.ipaddress))
-        tries = 0
-        while tries < 3:
+    def ssh_into_VM(self, vm, public_ip, reconnect=True, negative_test=False):
+        self.debug("SSH into VM with ID - %s on public IP address - %s" %
+                   (vm.id, public_ip.ipaddress.ipaddress))
+        tries = 1 if negative_test else 3
+        while True:
             try:
-                ssh_client = vm.get_ssh_client(ipaddress=public_ip.ipaddress.ipaddress, reconnect=reconnect)
+                ssh_client = vm.get_ssh_client(
+                    ipaddress=public_ip.ipaddress.ipaddress,
+                    reconnect=reconnect,
+                    retries=3 if negative_test else 30
+                )
+                self.debug("Successful to SSH into VM with ID - %s on "
+                           "public IP address - %s" %
+                           (vm.id, public_ip.ipaddress.ipaddress))
+                return ssh_client
             except Exception as e:
                 self.debug("Failed to SSH into VM: %s" % e)
-                self.debug("Waiting for the VM to be fully resolved for SSH connection...")
-                time.sleep(120)
+                tries -= 1
+                if tries == 0:
+                    raise e
```

Comment:
Please consider separating the retry logic into a utility function to make the code more readable and promote reuse.